tools/read_collection.cpp

```diff
@@ -4,54 +4,79 @@
 #include <range/v3/view/iota.hpp>
 
 #include "binary_collection.hpp"
+#include "io.hpp"
+#include "memory_source.hpp"
+#include "payload_vector.hpp"
 #include "util/util.hpp"
 
 using namespace pisa;
 
 int main(int argc, char** argv)
 {
     std::string collection_file;
-    std::string map_file{};
+    std::optional<std::string> map_file{};
+    std::optional<std::string> lex_file{};
     std::ptrdiff_t first, last;
 
-    CLI::App app{"read_collection - read collections."};
+    CLI::App app{"Reads binary collection to stdout."};
     app.add_option("-c,--collection", collection_file, "Collection file name")->required();
-    app.add_option("-m,--map", map_file, "String map file name");
-    app.add_option("first", first, "Element number")->required();
-    app.add_option("last", last, "Element number");
+
+    auto maptext =
+        app.add_option("--maptext", map_file, "Text file mapping each line number (ID) to a string");
```

Comment:
Maybe we could say "DocId" here just to be a bit more clear?

Reply:
Actually, this is not always DocId, as it works with both fwd and inv indexes, and in general with any "binary collection". But I agree, maybe we should be more verbose here:
```
ID to string mapping in text file format. Line n is the string associated with ID n.
E.g., if used to read a document from a forward index, this would be the `.terms`
file, which maps term IDs to their string reperesentations.
```

Same for lexicon:
```
ID to string mapping in lexicon binary file format.
E.g., if used to read a document from a forward index, this would be the `.termlex`
file, which maps term IDs to their string reperesentations.
```